api/v3/Case.php

```diff
@@ -645,23 +690,26 @@ function _civicrm_api3_case_read(&$cases, $options) {
  * @param array $params
  */
 function _civicrm_api3_case_format_params(&$params) {
-  // figure out case type id from case type and vice-versa
-  $caseTypes = CRM_Case_PseudoConstant::caseType('name', FALSE);
-  if (empty($params['case_type_id'])) {
-    $params['case_type_id'] = array_search($params['case_type'], $caseTypes);
-
-    // DEPRECATED: lookup by label for backward compatibility
-    if (!$params['case_type_id']) {
-      $caseTypeLabels = CRM_Case_PseudoConstant::caseType('title', FALSE);
-      $params['case_type_id'] = array_search($params['case_type'], $caseTypeLabels);
+  if (!empty($params['case_type_id']) || !empty($params['case_type'])) {
+    // figure out case type id from case type and vice-versa
+    $caseTypes = CRM_Case_PseudoConstant::caseType('name', FALSE);
+    if (empty($params['case_type_id'])) {
+      $params['case_type_id'] = array_search($params['case_type'], $caseTypes);
+
+      // DEPRECATED: lookup by label for backward compatibility
+      if (!$params['case_type_id']) {
+        $caseTypeLabels = CRM_Case_PseudoConstant::caseType('title', FALSE);
+        $params['case_type_id'] = array_search($params['case_type'], $caseTypeLabels);
+        $params['case_type'] = $caseTypes[$params['case_type_id']];
+      }
+    }
+    elseif (empty($params['case_type'])) {
```

Comment:
Early returns would improve this method's readability and considerably reduce the indentation level in this method